src/lexer.js

```diff
@@ -29,7 +33,17 @@ export function parse (source, name = '@') {
   return { exports: [...exports], reexports: [...reexports] };
 }
 
-function copy (src, outBuf16) {
+function copyBE (src, outBuf16) {
+  const len = src.length / 2;
+  let i = 0;
+  while (i < len) {
+    outBuf16[i] = src.charCodeAt(i * 2 + 1);
+    outBuf16[i + 1] = src.charCodeAt(i * 2);
```

Comment:
This doesn't appear to work as 
- `outBuf16[i + 1]` gets overwritten the next time around the loop by `outBuf16[i]`
- it swaps pairs of 16-bit values rather than the octets within the 16-bit value.

I've hacked something together that appears to work on AIX (tests pass). I'll need to tidy up a bit (I've almost certainly broken little endian in what I currently have) -- I'll try to get a PR in either later tonight or tomorrow.

Reply:
Thanks @richardlau, I've pushed up some fixes here, running on https://github.com/nodejs/node/pull/35634 to see if it works, but if it's still failing your PR would be a huge help.